### PR TITLE
workaround tornado py38 compatibility issue

### PR DESCRIPTION
**Pull request by echarles**
This is a port of https://github.com/jupyter/notebook/pull/5047

### CODE REVIEW

File: jupyter_server/serverapp.py [hunk 1]
```diff
@@ -1606,6 +1606,33 @@ def init_httpserver(self):
                               'no available port could be found.'))
             self.exit(1)
 
+    def _init_asyncio_patch(self):
```
Comment (Zsailer) [inline jupyter_server/serverapp.py:1609]: I think we can make this a `staticmethod` and call this in conftest—thereby removing the copy in conftest. I'll make these additions myself

File: tests/conftest.py [hunk 2]
```diff
@@ -64,7 +107,8 @@ def environ(
     data_dir,
     config_dir,
     runtime_dir,
-    root_dir
+    root_dir,
+    asyncio_patch
```
Comment (Zsailer) [inline tests/conftest.py:112]: I don't think we need to use this fixture in the environ. It will be picked up by the serverapp fixture.

**Comment by Zsailer** [comment]
This looks good. I'll push a couple of changes that I mentioned and merge.

File: tests/conftest.py (hunk 2 above)
Comment (echarles) [reply tests/conftest.py:112]: yep with my last commit it should not be needed here.

**Pull request merged**